src/cargo/util/lints.rs

```diff
@@ -880,6 +882,111 @@ pub fn unused_dependencies(
     Ok(())
 }
 
+pub fn unexpected_target_cfgs(
+    ws: &Workspace<'_>,
+    pkg: &Package,
+    path: &Path,
+    error_count: &mut usize,
+    gctx: &GlobalContext,
+) -> CargoResult<()> {
+    fn warn_on_unexpected_cfgs(
```

Comment:
Stylistically, we are not using inner function for any of our other lints that do similar processing and ... that argument is a big red flag to me.

Reply:
Yeah, I also realized that none of the lints are using inner functions, but putting this function at the same level felt even weirder. What about putting all of this logic in a submodule?